ACPs/113-provable-randomness/README.md

```diff
@@ -0,0 +1,157 @@
+
+Avalanche's traditional reliance on external oracles for randomness creates complexity and bottlenecks. These oracles inflate costs, hinder transaction speed, and are cumbersome to integrate. As Avalanche scales to more Subnets, this dependence on external systems becomes increasingly unsustainable.
+
+A solution for verifiable random number generation within Avalanche solves these problems. It provides fair randomness functionality across the chains, at no additional cost. This paves the way for a more efficient Avalanche ecosystem.
+
+## Specification
+
+### Changes Summary
+
+Existing avalanche protocol breaks the block building into two parts : external and internal. The external block is the SnowMan++ block, whereas the internal block is the actual virtual machine block.
+
+To support randomness, a BLS based VRF implementation is used, that would be recursively signing its own signatures as its message. Since the BLS signatures are deterministic, they provide a great way to construct a reliable VRF.
+
+For proposers that do not have a BLS key associated with their node, the hash of the signature from the previous round.
+
+In order to bootstrap the signatures chain, a missing signature would be replaced with a byte slice that is the hash product of a verifiable and trustable seed.
+
+The changes proposed here would affect the way a block is being validated. Therefore, when this change gets implemented, it needs to be deployed as a mandatory upgrade.
+
+```
+		+-----------------------+           +-----------------------+
+		|  Block n              | <-------- |  Block n+1            |
+		+-----------------------+           +-----------------------+
+		|  VRF-Sig(n)           |           |  VRF-Sig(n+1)         |
+		|  ...                  |           |  ...                  |
+		+-----------------------+           +-----------------------+
+
+		+-----------------------+           +-----------------------+
+		|  VM n                 |           |  VM n+1               |
+		+-----------------------+           +-----------------------+
+		|  VRF-Out(n)           |           |  VRF-Out(n+1)         |
+		+-----------------------+           +-----------------------+
+
+		VRF-Sig(n+1) = Sign(VRF-Sig(n), Block n+1 proposer's BLS key)
+		VRF-Out(n) = Hash(VRF-Sig(n))
+```
+
+### Changes Details
+
+#### Step 1. Adding BLS signature to proposed blocks
+
+```golang
+type statelessUnsignedBlock struct {
+	…
+	vrfSig    []byte `serialize:”true”`
+}
+```
+
+#### Step 2. Populate signature
+
+When a block proposer attempts to build a new block, it would need to use the parent block as a reference.
+
+The `vrfSig` field within each block is going to be daisy-chained to the `vrfSig` field from it's parent block.
+
+Populating the `vrfSig` would following this logic:
+
+1. The current proposer has a BLS key
+   
+	a. If the parent block has an empty `vrfSig` signature, that signature would be signed using the proposer’s BLS key. This is the base case.
```

Comment:
why sign the signature of the previous block and not sign the hash of the previous block?

Reply:
If you sign the hash of the previous block instead, I think it makes this document more compact / succinct as no need to specify special conditions to whether the previous block has the VRF signature or not.